File: src/main/java/com/eager/questioncloud/review/QuestionReviewJpaRepository.java
package com.eager.questioncloud.review;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface QuestionReviewJpaRepository extends JpaRepository<QuestionReviewEntity, Long> {
    Optional<QuestionReviewEntity> findByQuestionIdAndReviewerIdAndIsDeletedFalse(Long questionId, Long reviewerId);
}

File: qc-infra/qc-rdb/src/main/kotlin/com/eager/questioncloud/review/repository/QuestionReviewJpaRepository.kt
package com.eager.questioncloud.review.repository

import com.eager.questioncloud.review.entity.QuestionReviewEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface QuestionReviewJpaRepository : JpaRepository<QuestionReviewEntity, Long> {
    fun findByQuestionIdAndReviewerIdAndIsDeletedFalse(
        questionId: Long,
        reviewerId: Long
    ): Optional<QuestionReviewEntity>
    
    fun findByIdAndReviewerIdAndIsDeletedFalse(id: Long, reviewerId: Long): Optional<QuestionReviewEntity>
}
